pkg/artifactory/resource/security/resource_artifactory_group.go

```diff
@@ -5,257 +5,386 @@ import (
+
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to Delete Resource",
+			"An unexpected error occurred while attempting to delete the resource. "+
+				"Please retry the operation or report this issue to the provider developers.\n\n"+
+				"HTTP Error: "+err.Error(),
+		)
+
+		return
+	}
+
+	// Return error if the HTTP status code is not 200 OK or 404 Not Found
+	if response.StatusCode() != http.StatusNotFound && response.StatusCode() != http.StatusOK {
+		resp.Diagnostics.AddError(
+			"Unable to Delete Resource",
+			"An unexpected error occurred while attempting to delete the resource. "+
+				"Please retry the operation or report this issue to the provider developers.\n\n"+
+				"HTTP Status: "+response.Status(),
+		)
+
+		return
 	}
-	return diag.FromErr(err)
-}
 
-func resourceGroupExists(d *schema.ResourceData, m interface{}) (bool, error) {
-	return groupExists(m.(utilsdk.ProvderMetadata).Client, d.Id())
+	// If the logic reaches here, it implicitly succeeded and will remove
+	// the resource from state if there are no other errors.
 }
 
-func groupExists(client *resty.Client, groupName string) (bool, error) {
-	resp, err := client.R().Head(GroupsEndpoint + groupName)
-	if err != nil && resp != nil && resp.StatusCode() == http.StatusNotFound {
-		// Do not error on 404s as this causes errors when the upstream user has been manually removed
-		return false, nil
+// ImportState imports the resource into the Terraform state.
+func (r *ArtifactoryGroupResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+
+}
+func (r *ArtifactoryGroupResourceModel) ToState(ctx context.Context, group *ArtifactoryGroupResourceAPIModel, includeUsers bool) {
+	r.Id = types.StringValue(group.Name)
+	r.Name = types.StringValue(group.Name)
+	if group.Description != "" {
+		r.Description = types.StringValue(group.Description)
+	}
+	if group.ExternalId != "" {
+		r.ExternalId = types.StringValue(group.ExternalId)
 	}
+	r.AutoJoin = types.BoolValue(group.AutoJoin)
+	r.AdminPrivileges = types.BoolValue(group.AdminPrivileges)
+	r.Realm = types.StringValue(group.Realm)
+	if group.RealmAttributes != "" {
+		r.RealmAttributes = types.StringValue(group.RealmAttributes)
+	}
+	// We have to check if the value is not null to prevent an error "...unexpected new value: .users_names: was null, but now cty.SetValEmpty(cty.String)."
+	if !r.UsersNames.IsNull() {
+		if group.UsersNames != nil {
+			r.UsersNames, _ = types.SetValueFrom(ctx, types.StringType, group.UsersNames)
+		}
+	}
+	r.WatchManager = types.BoolValue(group.WatchManager)
+	r.PolicyManager = types.BoolValue(group.PolicyManager)
+	r.ReportsManager = types.BoolValue(group.ReportsManager)
+}
+
+func unableToCreateResourceError(resp *resource.CreateResponse, err error) {
+	resp.Diagnostics.AddError(
+		"Unable to Update Resource",
```

Comment:
Typo? s/Update/Create

Otherwise this func is identical to the one below.

Reply:
Fixed.